src/main/java/org/xbill/DNS/SVCBBase.java

```diff
@@ -185,20 +204,20 @@ public String toString() {
   }
 
   public static class ParameterAlpn extends ParameterBase {
-    private List<byte[]> values;
+    private final List<byte[]> values;
 
     public ParameterAlpn() {
       super();
       values = new ArrayList<>();
     }
 
-    public ParameterAlpn(List<byte[]> values) {
+    public ParameterAlpn(List<String> values) {
       super();
-      this.values = values;
+      this.values = values.stream().map(String::getBytes).collect(Collectors.toList());
     }
 
-    public List<byte[]> getValues() {
-      return values;
+    public List<String> getValues() {
+      return values.stream().map(String::new).collect(Collectors.toList());
```

Comment:
new uses the platform default encoding. This is often UTF-8, but it's definitely not guaranteed, so unless it's always ASCII, there will be problems.
(Also for getBytes, Unicode can come from the application and would then result in a mess in binary)

Reply:
Good point.  I made another commit to change this to s.getBytes(StandardCharsets.UTF_8) and new String(b, StandardCharsets.UTF_8).  Is that what you were suggesting?

Reply:
Well, yes, if the RFC/draft actually uses UTF-8 for the encoding of ALPNs?

Not so important, but I'd probably store the values as List<String> rather than List<byte[]> and convert in fromWire()/toWire()

Reply:
Upon further reading, I think it is limited to just ASCII.  So I will switch to that.

As for storing as a List<byte[]>, I was following the pattern used for other record types, such a TXTBase and AAAARecord, that internally store data this way, presumably to be closer to the wire format.